INSTALL.md

```diff
@@ -38,10 +38,22 @@ Dangerzone is available for:
   ```
 </details>
 
-Add our repository following [these instructions](https://packagecloud.io/firstlookmedia/code/install#manual-deb), or by running this script:
-
-```
-curl -s https://packagecloud.io/install/repositories/firstlookmedia/code/script.deb.sh | sudo bash
+Add our repository following these instructions:
+
+```sh
+sudo apt-get update
+sudo apt-get install -y curl
+sudo mkdir -p /etc/apt/keyrings/
+curl -S https://packages.freedom.press/apt-tools-prod/fpf-archive-keyring.gpg \
+    | sudo tee >/dev/null /etc/apt/keyrings/fpf.gpg
+source /etc/os-release
+echo deb [signed-by=/etc/apt/keyrings/fpf.gpg] https://packages.freedom.press/apt-tools-prod ${VERSION_CODENAME?} main \
+    | sudo tee /etc/apt/sources.list.d/fpf.list
+cat <<EOF | sudo tee /etc/apt/preferences.d/fpf.pref
+Package: *
+Pin: origin packages.freedom.press
+Pin-Priority: 100
+EOF
```

Comment:
can you explain a bit the logic behind the setting a priority? I was reading up a bit [on apt priorities](https://manpages.debian.org/buster/apt/apt_preferences.5.en.html) but I don't fully get yet what it mitigates here.

Reply:
Here's a bit more context: https://wiki.debian.org/DebianRepository/UseThirdParty#Standard_pinning

In a nutshell, you don't want an external repo to ship, e.g., their own `libc` and override the official one, because that's a huge security risk. With APT preferences, we can let external repos provide new packages, but have lower priority (100 < 500) for any other package.

Reply:
I understand. To be honest, I don't think this is a risk worth considering given that there already other ways an attacker controlling the repo can already do on the system. If my understanding is correct `post-install` scripts runs as root. So any attacker-controlled package could already do severe damage on the system without the need to ship a libc.

The line of defense that we have here is the fact that a webserver compromise won't result in malicious package installs since they won't control the signing keys.

Does this analysis make sense?

Reply:
It does make sense, and it's a point that a Debian developer also makes: https://blog.jak-linux.org/2021/06/20/migrating-away-apt-key/#pretending-to-be-safer-by-using-signed-by

The Debian community does not have a definitive answer to this, but it [acknowledges this problem](https://wiki.debian.org/DebianRepository/UseThirdParty):

> Note that the procedures documented here aim to prevent a repository from shipping packages that the administrator does not expect that repository to ship. For example, a repo that ships a video game emulator and its mods shouldn't be able to override [libc6](https://packages.debian.org/libc6).
> 
> However, the installation of any single malicious package from a malicious repository can currently undo these protections, for example by running a [MaintainerScripts](https://wiki.debian.org/MaintainerScripts) command to override the configured preferences or by authorizing new OpenPGP certificates. For the purposes of this page, attacks by a package that belongs to a given repository are out of scope. To restrict what an installed package can do, see the larger [UntrustedDebs](https://wiki.debian.org/UntrustedDebs) problem, and particularly [Teams/Dpkg/Spec/DeclarativePackaging](https://wiki.debian.org/Teams/Dpkg/Spec/DeclarativePackaging) for a potential solution.

---

In any case, we can remove this extra instruction, since it doesn't offer much more protection than pinning the GPG key. We can also explain to the user that:

* our instructions follow the [official Debian guide on external repos](https://wiki.debian.org/DebianRepository/UseThirdParty), and
* packages from external repos are installed with admin privileges, which is why we take several precautions to make sure that a malicious third-party cannot infect our packages.freedom.press infrastructure.

Reply:
OK. Let's remove it, then.